tests/type-annotations/main.fmf

```diff
@@ -0,0 +1,53 @@
+summary: Check type annotations in tmt code
+duration: 10m
+environment:
+    LANG: en_US.UTF-8
+
+    WITH_SYSTEM_PACKAGES: "no"
+
+tier: 0
+
+adjust+:
+  - when: distro == centos-stream-9
+    environment+:
+      HATCH_ENVIRONMENT: dev-not-editable
+
+require+:
```

Comment:
I guess we could use `make develop` directly. However, it's a bit unsatisfactory solution, instead of a declaration of packages required by a test, we'd instead take action and run `make`. I'm going to switch to this `make develop` command for now, but maybe we need to save these requirements somewhere from which both test and make could read it.